### PR TITLE
Adds Truck and rearEnd portion to body

### PR DESCRIPTION
**Pull request by Catyell2**
UML slices to follow soon but my slices are currently the `Truck` and `RearEnd` that will go into the `Body` that we are working on for class.

### CODE REVIEW

**Comment by Catyell2** [comment]
image


image

**Comment by Brandonhenk** [comment]
@Catyell2 Here is the update. Im not sure on what type the cabin would be in ExtendedCab and CrewCab, I asked Anthony waiting for a reply, I figured the standard truck would have a standard cabin. If you know what it would be then go for it and let me know.
body 2 - page 1-14

**Comment by Brandonhenk** [comment]
Update for RearEnd
bodycomponents - page 1-9

**Comment by Brandonhenk** [comment]
Got an answer from Anthony here is the update.
body 2 - page 1-15

**Comment by Catyell2** [comment]
Ok I've got a little trouble reading UML but that kinda makes sense that
they would be more luxury cabins. Talk on it tomorrow.

**Comment by Brandonhenk** [comment]
Updated both (should be last time) 
bodycomponents - page 1-2
body 2 - page 1-2

**Comment by Brandonhenk** [comment]
Made a small change to this so it makes more sense. Changed the name to Trunk. Also might be making more changes to the truck. I will keep you updated you can still code it it will be small changes.
bodycomponents - page 1-3

**Comment by Brandonhenk** [comment]
Here our the changes i made to truck. Here is the problem I dont know the answer to how all this should be. From what I know this makes sense but i could be wrong. 
body 2 - page 1-4

**Comment by Catyell2** [comment]
Alright sounds like a plan.

**Comment by anthonykulis** [comment]
Negative.  A truck bed is not a trunk. They are two different things and are not related hence should not be in the same inheritance line. You should have a class `TruckBed` in which `ShortBed` and `LongBed` inherit from, and you should have a class `Trunk` in which, very loosely, `Hatchback` inherits from. You have no need for a `StandardTrunk` class. Just make the `Trunk` instantiable.

File: edu/jalc/automobile/parts/LongBed.java [hunk 1]
```diff
@@ -0,0 +1,20 @@
+package edu.jalc.automobile.parts;
+
+private class LongBed extends TruckBed{
```
Comment (anthonykulis) [inline edu/jalc/automobile/parts/LongBed.java]: If this is private, no one can use it.

File: edu/jalc/automobile/parts/ShortBed.java [hunk 2]
```diff
@@ -0,0 +1,20 @@
+package edu.jalc.automobile.parts;
+
+private class ShortBed extends TruckBed{
```
Comment (anthonykulis) [inline edu/jalc/automobile/parts/ShortBed.java]: If this is private, no one can use it.

**Comment by Brandonhenk** [comment]
I don't understand. I do somewhat but everytime i make a change Im still wrong. Im getting no help with the UML I've spend many hours working on it just to keep getting it wrong I have other school work i have to do, and I can't keep changing Rick and James UML, they will never be able to code it.

**Comment by anthonykulis** [comment]
Sorry. I am pushy, no? I gave it thumbs up, so you can keep it as is. But the point behind building the domain model is to build something that is pretty obvious to anyone coming in to use it. They shouldn't have to stumble to find things. If you think they will not, then its good to go. If you think finding truck bed inside the trunk makes sense, keep it. I could be wrong. Its a constant negotiation and debate. When we move onto the next part, if its opaque to the team using it, and you find yourself explaining it to other teams, then we know it was too confusing.

**Comment by Catyell2** [comment]
Hey Brandon, I am going to try to pull down the `Cabin` things that I need to have for my tests.

**Comment by Brandonhenk** [comment]
I will take a look at it all later tonight I was busy and now I have to start some other work but after that I will take a look. I went ahead and made all the Seats myself. So I had those available for me and it still didn't work. I'm not sure I know how to fix it either.

Sent from my iPhone

On Mar 23, 2017, at 5:42 PM, James Jessup <notifications@github.com<mailto:notifications@github.com>> wrote:


Those broken tests are (I think) because of not having some parts of my Truck like the Seats and a similar problem should be occurring with the SUV too. I am also not super confident in my ability to write test.

**Comment by Catyell2** [comment]
Ok, not sure what we are going to do but talk with Anthony tomorrow for
sure. We definitely have a circular deal going on. We've done what we can
but we still aren't done, very frustrating.

**Comment by Brandonhenk** [comment]
Updated UML for James
bodycomponents - page 1-4
body 2 - page 1-5

**Comment by Catyell2** [comment]
body_2 - page 1 1

Truck looks more like this.

**Pull request merged**